cartridge/lua-api/compression.lua

```diff
@@ -134,73 +135,96 @@ function _G.__cartridgeGetStorageCompressionInfo(_)
-                                index_space:insert(multipart_key)
-                                table.insert(multipart_key, tuple[field_format_key])
-                                uncompressed_space:insert(multipart_key)
-                                compressed_space:insert(multipart_key)
-                                added = added + 1
-                            end
-                        end
+                local uncompressed_space =
+                    create_tmp_space(space_name, '_tmp_uncompressed',
+                        unique_index, space:format(), field_format)
+                field_format.compression = 'zstd' -- zstd lz4
+                local compressed_space =
+                    create_tmp_space(space_name, '_tmp_compressed',
+                        unique_index, space:format(), field_format)
+                local index_space =
+                    create_tmp_space(space_name, '_tmp_index',
+                        unique_index, space:format(), nil)
+
+                local random_seed = 0
+                local added = 0
+                local temp_space_len = space:len()
+                local temp_space_len_limit = 2000
+                if temp_space_len > temp_space_len_limit then
+                    temp_space_len = temp_space_len_limit
+                end
+
+                -- fill temporary spaces with limited count of items
+                while added < temp_space_len do
+                    random_seed = random_seed + 1
+                    local full_tuple = unique_index:random(random_seed)
+                    local tmp_tuple = {}
+                    for _, index_part in pairs(unique_index.parts) do
+                        local key_field = full_tuple[index_part.fieldno]
+                        table.insert(tmp_tuple, key_field)
+                    end
 
-                        local field_compression_info = {
-                            field_name = field_format.name,
-                            compression_percentage =
-                                (compressed_space:bsize() - index_space:bsize()) * 100 /
-                                (uncompressed_space:bsize() - index_space:bsize() + 1),
-                        }
-                        table.insert(space_compression_info, field_compression_info)
-
-                        compressed_space:drop()
-                        uncompressed_space:drop()
-                        index_space:drop()
+                    local exist_in_compressed_space = compressed_space:get(tmp_tuple)
+                    if exist_in_compressed_space == nil then
+                        index_space:insert(tmp_tuple)
+                        fiber.yield()
+                        table.insert(tmp_tuple, full_tuple[fieldno])
+                        uncompressed_space:insert(tmp_tuple)
+                        fiber.yield()
+                        compressed_space:insert(tmp_tuple)
+                        fiber.yield()
```

Comment:
I propose to put it to a single transaction (or at least the first two inserts) and test if there is any performance difference.

Reply:
So I had tested case with inserts in transaction. But there is no performance.